pipelines/pipeline_args.go

```diff
@@ -89,3 +91,39 @@ func PipelineArgsFromContext(ctx context.Context, c cliutil.CLIContext) (Pipelin
 		NPMOpts:          containers.NPMOptsFromFlags(c),
 	}, nil
 }
+
+// InjectPipelineArgsIntoSpan is used to copy some of the arguments passed to
+// the pipeline into a top-level OpenTelemtry span. Fields that might contain
+// secrets are left out.
+func InjectPipelineArgsIntoSpan(span trace.Span, args PipelineArgs) {
+	attributes := make([]attribute.KeyValue, 0, 10)
+	attributes = append(attributes, attribute.String("platform", string(args.Platform)))
+	if args.GrafanaOpts != nil {
+		if args.GrafanaOpts.GoVersion != "" {
+			attributes = append(attributes, attribute.String("go-version", args.GrafanaOpts.GoVersion))
+		}
+		if args.GrafanaOpts.Version != "" {
+			attributes = append(attributes, attribute.String("version", args.GrafanaOpts.Version))
+		}
+		if args.GrafanaOpts.GrafanaDir != "" {
+			attributes = append(attributes, attribute.String("grafana-dir", args.GrafanaOpts.GrafanaDir))
+		}
+		if args.GrafanaOpts.GrafanaRef != "" {
+			attributes = append(attributes, attribute.String("grafana-ref", args.GrafanaOpts.GrafanaRef))
+		}
+		if args.GrafanaOpts.EnterpriseDir != "" {
+			attributes = append(attributes, attribute.String("enterprise-dir", args.GrafanaOpts.EnterpriseDir))
+		}
+		if args.GrafanaOpts.EnterpriseRef != "" {
+			attributes = append(attributes, attribute.String("enterprise-ref", args.GrafanaOpts.EnterpriseRef))
+		}
```

Comment:
I almost think it'd be easier to query in the future if we added these attributes even with their default zero-values.

Reply:
Done :)